pkg/e2e/harness_runner/harness_runner.go

```diff
@@ -7,34 +7,114 @@ import (
+	harnesses         = strings.Split(viper.GetString(config.Tests.TestHarnesses), ",")
+	h                 *helper.H
+	err               error
+	HarnessEntries    []ginkgo.TableEntry
 )
 
 var _ = ginkgo.Describe("Test Harness", ginkgo.Ordered, label.TestHarness, func() {
-	var h *helper.H
-	ginkgo.BeforeAll(func() {
-		h = helper.New()
-	})
-
-	TimeoutInSeconds := viper.GetFloat64(config.Tests.PollingTimeout)
-	ginkgo.It("should run until completion", func(ctx context.Context) {
-		ginkgo.GinkgoWriter.Write([]byte("Test harness timeout is " + fmt.Sprintf("%v", TimeoutInSeconds)))
-		h.SetServiceAccount(ctx, viper.GetString(config.Tests.TestUser))
-		harnesses := strings.Split(viper.GetString(config.Tests.TestHarnesses), ",")
-		failed := h.RunTests(ctx, "test-harness", int(TimeoutInSeconds), harnesses, []string{})
-		if len(failed) > 0 {
-			message := fmt.Sprintf("Tests failed: %v", failed)
-			if url, ok := prow.JobURL(); ok {
-				message += "\n" + url
-			}
-			if err := alert.SendSlackMessage(viper.GetString(config.Tests.SlackChannel), message); err != nil {
-				log.Printf("Failed sending slack alert for test failure: %v", err)
-			}
-		}
-	}, TimeoutInSeconds+30)
+	for _, harness := range harnesses {
+		HarnessEntries = append(HarnessEntries, ginkgo.Entry("should run "+harness+" successfully", harness))
+	}
+	ginkgo.DescribeTable("Executing Harness",
+		func(harness string) {
+			ginkgo.By("======= RUNNING HARNESS: " + harness + " =======")
+			log.Printf("======= RUNNING HARNESS: %s =======", harness)
+			viper.Set(config.Project, "")
+			// Run harness in new project
+			h = helper.New()
+			h.SetServiceAccount(context.TODO(), "system:serviceaccount:%s:cluster-admin")
+			harnessImageIndex := strings.LastIndex(harness, "/")
+			harnessImage := harness[harnessImageIndex+1:]
+			suffix := util.RandomStr(5)
+			jobName := fmt.Sprintf("%s-%s", harnessImage, suffix)
+			r := h.RunnerWithNoCommand()
+			testCommand, err := getCommandString(h, harness, r, suffix, jobName)
+			Expect(err).NotTo(HaveOccurred(), "Could not create test pod template")
+			r.Cmd = testCommand
+
+			// run tests
+			stopCh := make(chan struct{})
+			err = r.Run(int(TimeoutInSeconds), stopCh)
+			Expect(err).NotTo(HaveOccurred(), "Could not run pod")
+
+			// get results
+			results, err := r.RetrieveTestResults()
+			Expect(err).NotTo(HaveOccurred(), "Could not read results")
+
+			// write results
+			h.WriteResults(results)
+
+			// ensure job has not failed
+			_, err = h.Kube().BatchV1().Jobs(r.Namespace).Get(context.TODO(), jobName, metav1.GetOptions{})
+			Expect(err).NotTo(HaveOccurred(), "Harness job pods failed")
+
+			h.Cleanup(context.TODO())
+			ginkgo.By("======= FINISHED HARNESS: " + harness + " =======")
+		},
+		HarnessEntries)
 })
+
+// Generates templated command string to provide to test harness container
+func getCommandString(h *helper.H, harness string, r *runner.Runner, suffix string, jobName string) (string, error) {
```

Comment:
Call `ginkgo.GinkgoHelper()` at the top of this function so that ginkgo properly handles the offset and get rid of the error return. That way we don't both check errors with gomega internally, and return an error as well

Reply:
Still returning an error here